### PR TITLE
feature: Add GoSec support

### PR DESCRIPTION
**Pull request by rtfpessoa**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -56,6 +56,16 @@ inputs:
     description: >-
       Reduce issue severity by one level, for non-security issues, for compatibility with GitHub's code scanning feature.
       This option will only have an effect when used in conjunction with '--format sarif'.
+  run-docker-tools:
```
Comment (afsmeira) [inline action.yml:60]: This option should be documented.

**Comment by rtfpessoa** [comment]
Codacy seems to be having problems due to the white-space, but the lines seem fine.

**Review by mrfyda — APPROVED** [review]

File: action.yml [hunk 2]
```diff
@@ -79,21 +124,32 @@ runs:
...
+            $(if [ ${{ inputs.upload }} = true ]; then echo "--upload"; fi)
+            $(if [ ${{ inputs.fail-if-incomplete }} = true ]; then echo "--fail-if-incomplete"; fi)
+            $(if [ ${{ inputs.allow-network }} = true ]; then echo "--allow-network"; fi)
+            $(if [ ${{ inputs.force-file-permissions }} = true ]; then echo "--force-file-permissions"; fi)
+            $(if [ ${{ inputs.gh-code-scanning-compat }} = true ]; then echo "--gh-code-scanning-compat"; fi)
+        else
+          echo "Skipping docker tools"
+        fi
+
+    - name: "Let Codacy know it can start processing the analysis results"
```
Comment (mrfyda) [inline action.yml:160]: run only if "inputs.upload" or "inputs.run-gosec"?

**Review by mrfyda — CHANGES_REQUESTED** [review]

File: action.yml (hunk 2 above)
Comment (rtfpessoa) [reply action.yml:160]: Interesting

**Review by mrfyda — DISMISSED** [review]

**Review by afsmeira — APPROVED** [review]

**Review by afsmeira — DISMISSED** [review]

File: action.yml [hunk 3]
```diff
@@ -56,15 +56,65 @@ inputs:
...
         echo "ORGANIZATION_PROVIDER=$(if [ "$GITHUB_SERVER_URL" == "https://github.com" ]; then echo "gh"; else echo "ghe"; fi)" >> $GITHUB_ENV
+
+    - name: "Prepare curl authentication header"
+      shell: bash
+      run: |
+        if [ -n "${{ inputs.api-token }}" ]; then
+          echo "CURL_CODACY_AUTH_AUTHENTICATION=api-token: ${{ inputs.api-token }}" >> $GITHUB_ENV
+        else
+          echo "CURL_CODACY_AUTH_AUTHENTICATION=project-token: ${{ inputs.project-token }}" >> $GITHUB_ENV
+        fi
```
Comment (afsmeira) [inline action.yml]: Both API token and project token are not required, so you can't do an `else` here, right?

File: action.yml [hunk 4]
```diff
@@ -80,21 +130,36 @@ runs:
...
+
+    - name: "Let Codacy know it can start processing the analysis results"
+      shell: bash
+      run: |
+        if [ ${{ inputs.upload }} = true ]; then
+          curl -XPOST -L -H "$CURL_CODACY_AUTH_AUTHENTICATION" \
+            -H "Content-type: application/json" \
+            "${CODACY_BASE_URL_OR_DEFAULT}/2.0/$ORGANIZATION_PROVIDER/$OWNER_NAME/$REPOSITORY_NAME/commit/$GITHUB_SHA/resultsFinal"
+        else
+          echo "Skipping upload"
```
Comment (afsmeira) [inline action.yml]: Maybe change to "Skipping results upload"

File: action.yml (hunk 3 above)
Comment (rtfpessoa) [reply action.yml]: one of them needs to be filled I guess I can improve this

File: action.yml [hunk 5]
```diff
@@ -56,15 +56,68 @@ inputs:
...
+
+    - name: "Prepare curl authentication header"
+      shell: bash
+      run: |
+        if [ -n "${{ inputs.api-token }}" ]; then
+          echo "CURL_CODACY_AUTH_AUTHENTICATION=api-token: ${{ inputs.api-token }}" >> $GITHUB_ENV
+        elif [ -n "${{ inputs.project-token }}" ]; then
+          echo "CURL_CODACY_AUTH_AUTHENTICATION=project-token: ${{ inputs.project-token }}" >> $GITHUB_ENV
+        else
+          echo "At least one authentication method is required."
```
Comment (afsmeira) [inline action.yml]: Wait, if you just want to run this locally, you can't.

Comment (rtfpessoa) [reply action.yml]: What do you mean running the action locally?

Comment (afsmeira) [reply action.yml]: I meant without uploading results. You don't really need a token to run the action.

Comment (rtfpessoa) [reply action.yml]: ok, i see. I changed it to only apply to upload

**Review by afsmeira — APPROVED** [review]

**Pull request merged**